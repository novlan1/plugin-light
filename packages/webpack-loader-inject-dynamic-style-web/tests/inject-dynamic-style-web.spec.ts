import * as fs from 'fs';
import path from 'path';
import * as loaderUtils from 'loader-utils';
import injectDynamicStyleWeb from '../src/loader';


jest.mock('fs');
jest.mock('loader-utils');

const dir = path.resolve(process.cwd(), 'src/local-component/ui/tip-match/tip-match-notice/css');
const dirWithoutComponentName = path.resolve(process.cwd(), 'src/local-component/ui/tip-match/tip-match-notice');

const source = `.tip-btn-primary{
    color: #212124;
    background-size: 100% 100%;
}`;

describe('injectDynamicStyleWeb', () => {
  const oldEnv = process.env;

  beforeAll(() => {
    jest.spyOn(fs, 'readdirSync').mockReturnValue(['base.scss', 'dialog.scss', 'icon.scss'] as any);
  });

  beforeEach(() => {
    process.env = { ...oldEnv };
    process.env.UNI_PLATFORM = 'h5';
  });

  afterEach(() => {
    process.env = oldEnv;
    jest.restoreAllMocks();
  });

  it('should not be used on non-H5 platform', () => {
    process.env.UNI_PLATFORM = 'mp-weixin';
    expect(injectDynamicStyleWeb.call({ context: dir }, source)).toBe(source);

    process.env.UNI_PLATFORM = 'mp-qq';
    expect(injectDynamicStyleWeb.call({ context: dir }, source)).toBe(source);
  });

  it('should not be used when failed to get the component name', () => {
    expect(injectDynamicStyleWeb.call({ context: dirWithoutComponentName }, source)).toBe(source);
  });

  it('should work correctly', () => {
    expect(injectDynamicStyleWeb.call({ context: dir }, source)).toMatchSnapshot();
  });

  it('should set topElement correctly', () => {
    jest.spyOn(loaderUtils, 'getOptions').mockReturnValue({ topElement: '#container' });
    expect(injectDynamicStyleWeb.call({ context: dir }, source)).toMatchSnapshot();
  });
});
