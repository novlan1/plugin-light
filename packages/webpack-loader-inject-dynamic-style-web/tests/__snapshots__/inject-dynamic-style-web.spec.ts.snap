// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`injectDynamicStyleWeb should set topElement correctly 1`] = `
".tip-btn-primary{
    color: #212124;
    background-size: 100% 100%;
}

#container.tip-match-notice {
  
  &--type-dialog {
    @import './dialog.scss';
  }

  &--type-icon {
    @import './icon.scss';
  }
}
"
`;

exports[`injectDynamicStyleWeb should work correctly 1`] = `
".tip-btn-primary{
    color: #212124;
    background-size: 100% 100%;
}

body.tip-match-notice {
  
  &--type-dialog {
    @import './dialog.scss';
  }

  &--type-icon {
    @import './icon.scss';
  }
}
"
`;
